File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.java
public class Solution {

    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int left = 0;
        int right = 0;
        Set<Integer> set = new HashSet<>();

        while (right < nums.length) {
            if (right - left <= k) {
                if (set.contains(nums[right])) {
                    return true;
                }
                set.add(nums[right]);
                right++;
            } else {
                set.remove(nums[left]);
                left++;
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var left = 0
        var right = 0
        var res = 1

        while (right < nums.size) {
            if (nums[left] == nums[right]) {
                right++
            } else {
                nums[++left] = nums[right]
                res++
            }
        }

        return res
    }
}